fixed-map-derive/src/unit_variants.rs

```diff
@@ -8,82 +8,53 @@ use crate::context::Ctxt;
+    let iterator_flatten = cx.toks.iterator_flatten();
+    let iterator_t = cx.toks.iterator_t();
+    let key_trait = cx.toks.key_t();
+    let mem = cx.toks.mem();
+    let occupied_entry_t = cx.toks.occupied_entry_t();
+    let option = cx.toks.option();
+    let option_bucket_none = cx.toks.option_bucket_none();
+    let option_bucket_option = cx.toks.option_bucket_option();
+    let option_bucket_some = cx.toks.option_bucket_some();
+    let ord_t = cx.toks.ord_t();
+    let ordering = cx.toks.ordering();
+    let partial_eq_t = cx.toks.partial_eq_t();
+    let partial_ord_t = cx.toks.partial_ord_t();
+    let slice_iter = cx.toks.slice_iter();
+    let slice_iter_mut = cx.toks.slice_iter_mut();
+    let storage_entry_t = cx.toks.storage_entry_t();
+    let storage_trait = cx.toks.storage_t();
+    let vacant_entry_t = cx.toks.vacant_entry_t();
 
     let const_wrapper = Ident::new(
         &format!("__IMPL_KEY_FOR_{}", cx.ast.ident),
         Span::call_site(),
     );
 
-    let mut len = Vec::new();
-    let mut is_empty = Vec::new();
-    let mut pattern = Vec::new();
-    let mut names = Vec::new();
-    let mut fields = Vec::new();
-    let mut field_inits = Vec::new();
-    let mut contains_key = Vec::new();
-    let mut get = Vec::new();
-    let mut get_mut = Vec::new();
-    let mut insert = Vec::new();
-    let mut remove = Vec::new();
-    let mut retain = Vec::new();
-    let mut keys_iter_init = Vec::new();
-    let mut iter_init = Vec::new();
-    let mut entry = Vec::new();
+    let count = en.variants.len();
+    let mut variants = Vec::with_capacity(count);
+    let mut names = Vec::with_capacity(count);
+    let mut field_inits = Vec::with_capacity(count);
 
     for (index, variant) in en.variants.iter().enumerate() {
-        let var = &variant.ident;
-        let name = Ident::new(&format!("f{}", index), Span::call_site());
-
-        len.push(quote!(usize::from(#option::is_some(#name))));
-        is_empty.push(quote!(#option::is_none(#name)));
         field_inits.push(quote!(#option::None));
-        fields.push(quote!(#option<V>));
-        pattern.push(quote!(#ident::#var));
-        contains_key.push(quote!(#option::is_some(#name)));
-        get.push(quote!(#option::as_ref(#name)));
-        get_mut.push(quote!(#option::as_mut(#name)));
-        insert.push(quote!(#mem::replace(#name, #option::Some(value))));
-        remove.push(quote!(#mem::take(#name)));
-        retain.push(quote! {
-            if let Some(val) = #option::as_mut(#name) {
-                if !func(#ident::#var, val) {
-                    *#name = None;
-                }
-            }
-        });
-        keys_iter_init.push(quote!(if #name.is_some() { Some(#ident::#var) } else { None }));
-        iter_init.push(quote!((#ident::#var, #name)));
-        names.push(name.clone());
-        entry.push(quote!(option_to_entry(#name, key)));
+        variants.push(&variant.ident);
+        names.push(format_ident!("_{}", index));
```

Comment:
Can we rename `variants` and `names` to just `variant` and `name`? 
Personally, I prefer `#ident::#variant` to `#ident::#variants`.
I feel like plurals should be reserved for cases where it's meant to be expanded directly into a list, like `#(#field_inits,)*`.

I think a better make than `#ident` would be great too. Maybe `#target`?

Reply:
:thinking: I don't have particularly strongly held opinions here. The larger improvements is reducing the number of vectors in use and moving the code closer to how it's used which makes it easier to work with.

One thought I have though is that pluralized names is useful to identify what's being looped over in the `quote!` macro call even though it looks a little awkward.